torch/_dynamo/variables/tensor.py

```diff
@@ -849,137 +849,93 @@ class TensorWithTFOverrideVariable(VariableTracker):
+        if var.global_class_name() not in tx.output.global_scope:
+            tx.output.install_global(var.global_class_name(), subclass_type)
+
         return var
 
     def __init__(
         self,
         tensor_variable,
-        orig_tensor_variable_source,
-        subclass_torch_function__func,
+        torch_function_fn,
         subclass_type,
         **kwargs,
     ):
         super().__init__(**kwargs)
         self.tensor_variable = tensor_variable
-        self.orig_tensor_variable_source = orig_tensor_variable_source
-        self.subclass_torch_function__func = subclass_torch_function__func
+        self.torch_function_fn = torch_function_fn
         self.subclass_type = subclass_type
 
-    def call_method(
-        self,
-        tx,
-        name,
-        args: "List[VariableTracker]",
-        kwargs: "Dict[str, VariableTracker]",
-    ) -> "VariableTracker":
-        # This code block implements inlining the __torch_function__ override
-        # of `call_method`.
-        from . import GetAttrVariable
+    def as_proxy(self):
+        return self.tensor_variable.as_proxy()
 
-        options = VariableTracker.propagate(self, args, kwargs.values())
-        # insert unwrapped version of self as the first argument
-        # TODO: This is wrong!  When you call the internal __torch_function__,
-        # you still get the wrapped version of self, and if you call functions
-        # inside __torch_function__, they should come back here.  If we unwrap
-        # the tensor immediately, that will not happen.
-        # See https://github.com/pytorch/torchdynamo/issues/1951
-        args = list(args)
-        args.insert(0, self.tensor_variable)
-        func_var = GetAttrVariable(self.tensor_variable, name)
-
-        unwrapped = TensorWithTFOverrideVariable.inline_torch_function_unwrapped(
-            tx,
-            func_var,
-            self.orig_tensor_variable_source,
-            self.subclass_torch_function__func,
-            self.subclass_type,
-            options,
-            args,
-            kwargs,
-        )
+    def python_type(self):
+        return self.subclass_type
 
-        # TODO(future PR): implement rewrapping conditional on method presence
-        # in `torch.overrides.get_default_nowrap_function()`. It's unclear how
-        # to do this easily in the current codebase since the resolution of
-        # `GetAttrVariable` depends on the type of the underlying object.
+    def torch_function_var(self, tx):
+        from .builder import VariableBuilder
 
-        return TensorWithTFOverrideVariable(
-            unwrapped,
-            self.orig_tensor_variable_source,
-            self.subclass_torch_function__func,
-            self.subclass_type,
+        source = AttrSource(
+            AttrSource(self.source, "__torch_function__"),
+            "__func__",
         )
+        return VariableBuilder(tx, source)(self.torch_function_fn)
```

Comment:
guards?

Reply:
As discussed I'll build this in the caller